ksql-engine/src/test/java/io/confluent/ksql/internal/KsqlEngineMetricsTest.java

```diff
@@ -93,72 +95,87 @@ public void shouldRemoveAllSensorsOnClose() {
     });
   }
 
+  private void shouldRecordMetric(final String name, final double expected) {
+    assertThat(getMetricValueLegacy(engineMetrics.getMetrics(), name), equalTo(expected));
+    assertThat(getMetricValue(engineMetrics.getMetrics(), name), equalTo(expected));
+  }
+
+  private void shouldRecordRate(final String name, final double expected, final double error) {
+    assertThat(
+        Math.floor(getMetricValueLegacy(engineMetrics.getMetrics(), name)),
+        closeTo(expected, error));
+    assertThat(
+        Math.floor(getMetricValue(engineMetrics.getMetrics(), name)),
+        closeTo(expected, error));
+  }
+
+  private void shouldRecordQueryCountMetric(final String name, long expected) {
+    assertThat(
+        getLongMetricValueLegacy(engineMetrics.getMetrics(), "testGroup-query-stats-" + name),
+        equalTo(expected)
+    );
+    assertThat(getLongMetricValue(engineMetrics.getMetrics(), name), equalTo(expected));
+  }
+
   @Test
   public void shouldRecordNumberOfActiveQueries() {
     when(ksqlEngine.numberOfLiveQueries()).thenReturn(3);
-    final double value = getMetricValue(engineMetrics.getMetrics(), metricNamePrefix + "num-active-queries");
-    assertEquals(3.0, value, 0.0);
+    shouldRecordMetric("num-active-queries", 3.0);
```

Comment:
The test is much more readable with everything in-line IMHO.

Consider in-lining, e.g. 

```java
// Given:
when(ksqlEngine.numberOfLiveQueries()).thenReturn(3);

// When:
final double value = getMetricValue(engineMetrics.getMetrics(), metricNamePrefix + "num-active-queries");

final double legacyValue = getMetricValueLegacy(engineMetrics.getMetrics(), name);

// Then:
assertThat(value, is(3.0));
assertThat(legacy, is(3.0));
```

Or even better - split into two tests: one for legacy and one for new. Then you can just delete the legacy ones when you remove that.   This, IMHO, is better as its good practice to have each test-case test only one thing.